libr/include/r_util/r_type.h

```diff
@@ -39,6 +39,26 @@ R_API int r_type_unlink(Sdb *TDB, ut64 addr);
 R_API int r_type_link_offset(Sdb *TDB, const char *val, ut64 addr);
 R_API char *r_type_format(Sdb *TDB, const char *t);
 
+// Builtin C type specifier parsing
+// Parses types like "unsigned long long", "uint32_t", "size_t", etc.
+// Returns true if the type is recognized, fills in size (bytes) and is_signed
+typedef enum {
+	R_TYPE_CTYPE_UNKNOWN = 0,
```

Comment:
uhm the problem with this enum is that for me there's a mixed confussion because signed, unsigned are attributes, but float is an unspecific type (there are tons of floating point standards) and then bool is a type. so i would prefer to have this cleanly implemented with type, base type and attributes, also for float we have an api to define every single bit possible too.